lib/src/builder/parsing/members/component.dart

```diff
@@ -119,15 +119,16 @@ class BoilerplateComponent extends BoilerplateMember {
     });
 
     // Ensure that Component2 declarations do not use legacy lifecycle methods.
-    if (isComponent2(version) && node is ClassOrMixinDeclaration) {
```

Comment:
Similar to the other comment, this check is a no-op for ClassTypeAlias, so excluding it here is unnecessary.